apps/src/radar.rs

```diff
@@ -290,16 +311,36 @@ fn main() {
                         let mut rows = vec![];
                         // make a vec of all strings to get a total amount of airplanes with
                         // position information
+                        let empty = "".to_string();
                         for key in adsb_airplanes.0.keys() {
-                            let value = adsb_airplanes.lat_long_altitude(*key);
-                            if let Some((position, altitude)) = value {
-                                rows.push(Row::new(vec![
-                                    format!("{}", key),
-                                    format!("{}", position.latitude),
-                                    format!("{}", position.longitude),
-                                    format!("{}", altitude),
-                                ]))
+                            let state = adsb_airplanes.0.get(key).unwrap();
+                            let pos = adsb_airplanes.lat_long_altitude(*key);
+                            let mut lat = empty.clone();
+                            let mut lon = empty.clone();
+                            let mut alt = empty.clone();
+                            if let Some((position, altitude)) = pos {
+                                lat = format!("{}", position.latitude);
+                                lon = format!("{}", position.longitude);
+                                alt = format!("{}", altitude);
                             }
+                            rows.push(Row::new(vec![
+                                format!("{}", key),
+                                state.callsign.as_ref().unwrap_or(&empty).clone(),
+                                lat,
+                                lon,
+                                format!("{:>8}", alt),
+                                if let Some(vert_speed) = state.vert_speed {
```

Comment:
I feel like there's a better way to do this that I'm missing.

Reply:
you could do:
```
state
    .vert_speed
    .map_or_else(|| "".to_string(), |v| format!("{:>6}", v)),
```

I'm not sure that's an improvement?